snapcraft/internal/pluginhandler.py

```diff
@@ -322,6 +322,13 @@ def clean_build(self, hint=''):
     def migratable_fileset_for(self, step):
         plugin_fileset = self.code.snap_fileset()
         fileset = (getattr(self.code.options, step, ['*']) or ['*']).copy()
+        # If we're priming and we don't have an explicit set of files to prime
+        # include the files from the stage step
+        if (step == 'prime' or step == 'snap') and fileset == ['*']:
```

Comment:
Ah, you're right.  I've switched it to 'prime' to be consistent with the calling method.